Source/ASVideoNode.mm

```diff
@@ -707,6 +710,15 @@ - (void)resetToPlaceholder
 
 #pragma mark - Playback observers
 
+- (void)applicationDidBecomeActive:(NSNotification *)notification
+{
+  ASDN::MutexLocker l(__instanceLock__);
+  
+  if (_shouldBePlaying && ASInterfaceStateIncludesVisible(self.interfaceState)) {
```

Comment:
We should avoid calling `-interfaceState` while holding the lock, since that method takes the lock. Let's remove the explicit lock, and just call `self.shouldBePlaying` to keep things simple. We can also use `self.visible` as a shorthand for `ASInterfaceStateIncludesVisible(self.interfaceState)`.